org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionResolveHandler.java

```diff
@@ -57,4 +82,111 @@ public CodeAction resolve(CodeAction params, IProgressMonitor monitor) {
 
 		return params;
 	}
+
+	private static final void addSnippetsIfApplicable(LinkedCorrectionProposalCore proposal, WorkspaceEdit edit) throws CoreException {
+		Object modifiedElement = proposal.getChange().getModifiedElement();
+		ICompilationUnit compilationUnit = (ICompilationUnit) ((IJavaElement) modifiedElement).getAncestor(IJavaElement.COMPILATION_UNIT);
+		IBuffer buffer = compilationUnit.getBuffer();
+		LinkedProposalModelCore linkedProposals = proposal.getLinkedProposalModel();
+		List<Triple> snippets = new ArrayList<>();
+		Iterator<LinkedProposalPositionGroupCore> it = linkedProposals.getPositionGroupCoreIterator();
+		int snippetNumber = 1;
+		while (it.hasNext()) {
+			LinkedProposalPositionGroupCore group = it.next();
+			ProposalCore[] proposalList = group.getProposals();
+			PositionInformation[] positionList = group.getPositions();
+			StringBuilder snippet = new StringBuilder();
+			snippet.append(SNIPPET_CHOICE_INDICATOR);
+			if (proposalList.length > 1) {
```

Comment:
As discussed, we should probably suport proposalList=1 , and even an empty proposalList to match the same support (ie. placeholders) that Eclipse has in these cases.